bin/e2e/remove-test-user.ts

```diff
@@ -0,0 +1,61 @@
+import { IdentityTypes } from "../../lib/@types";
+import { deleteUser } from "../../lib/commands/deleteUser";
+import { getUserFiltersByProvider } from "../../lib/queries/getUserFiltersByProvider";
+
+async function removeTestUser(): Promise<void> {
+  if (process.env.GITHUB_CONTEXT_EVENT === undefined) {
+    throw new Error("GITHUB_CONTEXT_EVENT environment variable is undefined");
+  }
+
+  const githubActionsContext = JSON.parse(process.env.GITHUB_CONTEXT_EVENT);
+
+  if (githubActionsContext.deployment === undefined) {
+    throw new Error("deployment is undefined");
+  }
+
+  if (githubActionsContext.deployment_status === undefined) {
+    throw new Error("deployment_status is undefined");
+  }
+
+  if (process.env.CONNECT_TEST_ACCOUNT_EMAIL === undefined) {
+    throw new Error(
+      "CONNECT_TEST_ACCOUNT_EMAIL environment variable is undefined",
+    );
+  }
+
+  const githubDeploymentStatus = githubActionsContext.deployment_status;
+
+  if (!githubDeploymentStatus.environment.includes("storybook")) {
+    if (githubDeploymentStatus.state === "success") {
+      const testUserEmail = process.env.CONNECT_TEST_ACCOUNT_EMAIL.split(
+        "@",
+      ).join("_" + githubActionsContext.deployment.sha + "@");
+
+      const testUserId = await getUserFiltersByProvider({
+        value: testUserEmail,
+        type: IdentityTypes.EMAIL,
```

Comment:
I think only using the value would be best, as our forms check for identity values format. We also want to avoid filter composition, from a management POV.